### PR TITLE
add subgraph, contract hooks, lend utils

### PR DESCRIPTION
**Pull request by timongll**

### CODE REVIEW

File: lend/src/hooks/useAssetPrice.ts [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+  const apiURL = `${COINGECKO_BASE_URL}/coins/${currencyName}/market_chart?vs_currency=usd&days=365&interval=daily`;
+
+  const response = await axios.get(apiURL);
+  const { data } = response;
+
+  return data.prices.map(([timestamp, price]: number[]) => ({
+    timestamp,
+    price,
+  }));
+};
```
Comment (leonkho) [inline lend/src/hooks/useAssetPrice.ts:29]: I wonder if this should be exported into the shared dir?

File: lend/src/hooks/useAssetPrice.ts [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+      };
+    },
+    searchAssetPriceFromDate,
+    searchAssetPriceFromMoment,
+    searchAssetPriceFromTimestamp,
+  };
+};
+
+interface AssetInfo {
+  circulating_supply: number;
```
Comment (leonkho) [inline lend/src/hooks/useAssetPrice.ts:247]: This should be camelcased

File: lend/src/utils/signing.ts [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+  yParityAndS: string;
+  compact: string;
+}
+export const domainType = [
+  { name: "name", type: "string" },
+  { name: "version", type: "string" },
+  { name: "chainId", type: "uint256" },
+];
+
+export function splitSignature(signature: ISignatureLike): ISignature {
```
Comment (leonkho) [inline lend/src/utils/signing.ts:34]: This should also be exported considering Earn is using the same util

File: lend/src/hooks/useAssetPrice.ts (hunk 1 above)
Comment (timongll) [reply lend/src/hooks/useAssetPrice.ts:29]: its the same as in the shared dir just that the Assets mapping in smaller so was thinking the array queries would be faster

**Review by leonkho — APPROVED** [review]

**Pull request merged**